### PR TITLE
Move TestWatcherStart to integration tests

### PR DESCRIPTION
**Pull request by Katharine**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Moves the test `TestStartingWatcher` from unit tests to integration tests. `TestStartingWatcher` depends on the system it runs on having a `/dev/kmsg`, which is not necessarily true — for instance, in an unprivileged container there is no such file. It is the only unit test to make an assumption along those lines

Given the purpose of the test is just to test whether a real watcher (which attempts to interact with the system) can be started, I think this reasonably qualifies as an integration test, rather than a test to be worked around by the environments the test runs in.

See e.g. [this test run](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-cached-make-test/1235664579724316673) for a failing test in an unprivileged container.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig node
/sig testing
/priority important-soon
/assign @mattjmcnaughton

### CODE REVIEW

**Comment by BenTheElder** [comment]
cc @liggitt

File: test/integration/kubelet/oom_watcher_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/kubelet/oom"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestStartingWatcher tests that the watcher, using the actual streamer
+// and not the fake, starts successfully.
+func TestStartingWatcher(t *testing.T) {
```
Comment (liggitt) [inline test/integration/kubelet/oom_watcher_test.go]: does this file need to retain the magic `_linux_test.go` suffix to remain linux only?

**Comment by BenTheElder** [comment]
pretty in favor of just removing this broken "unit" test in favor of e2e etc..

**Review by mattjmcnaughton — COMMENTED** [review]
I'm +1 with @BenTheElder that we may just want to delete this specific test.

We can add a more meaningful e2e test if we which, or just be comfortable with the current tests which mock out the `oomStreamer`.

**Comment by BenTheElder** [comment]
er I feel I should add: before this PR was filed while discussing this test I suggested move here or remove. I'm leaning towards remove now after talking to @liggitt but I also suggested this move so ... :sweat_smile:

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/90927
/close

**Pull request closed without merging**